nautobot/dcim/factory.py

```diff
@@ -711,3 +715,92 @@ class Params:
         lambda o: o.parent.controller if o.parent else Controller.objects.order_by("?").first()
     )
     weight = factory.Faker("pyint", min_value=1, max_value=1000)
+
+
+class ModuleTypeFactory(PrimaryModelFactory):
+    class Meta:
+        model = ModuleType
+        exclude = ("has_part_number",)
+
+    manufacturer = random_instance(Manufacturer, allow_null=False)
+
+    model = factory.LazyAttributeSequence(lambda o, n: f"{o.manufacturer.name} ModuleType {n + 1}")
+
+    has_part_number = NautobotBoolIterator()
+    part_number = factory.Maybe("has_part_number", factory.Faker("ean", length=8), "")
+
+
+class ModuleBayFactory(PrimaryModelFactory):
+    class Meta:
+        model = ModuleBay
+        exclude = (
+            "has_description",
+            "has_label",
+            "has_parent_module",
+        )
+
+    has_parent_module = False
+    parent_device = factory.Maybe("has_parent_module", None, random_instance(Device, allow_null=False))
+    parent_module = factory.Maybe("has_parent_module", random_instance(Module, allow_null=False), None)
+
+    position = factory.Maybe(
+        "has_parent_module",
+        factory.LazyAttribute(lambda o: o.parent_module.module_bays.count() + 1),
+        factory.LazyAttribute(lambda o: o.parent_device.module_bays.count() + 1),
+    )
+    has_label = NautobotBoolIterator()
+    label = factory.Maybe(
+        "has_label",
+        factory.Maybe(
+            "has_parent_module",
+            factory.LazyAttribute(lambda o: f"{o.parent_module!s} Bay {o.position}"),
+            factory.LazyAttribute(lambda o: f"{o.parent_device!s} Bay {o.position}"),
```

Comment:
Curious as to what syntax this is🤔

Reply:
The `!s` in the fstring?

Reply:
oops never mind, didn't see the f-string